### PR TITLE
Enable Jetpack search module UI

### PR DESCRIPTION
**Pull request by gravityrail**
Enable the new Jetpack search module toggle.

see: https://github.com/Automattic/jetpack/pull/7829

### CODE REVIEW

**Comment by beaulebens** [comment]
Works from a quick test (functionally). Haven't looked at the code yet. Here's what it looks like (the "activating search" message/changing description text is a bit of a weird pattern that I don't know we use anywhere else, we might want to just reword the descriptive text to always be relevant, and not have it change during activation):

screen shot 2017-09-25 at 1 44 58 pm
screen shot 2017-09-25 at 1 45 08 pm

**Comment by gravityrail** [comment]
> the "activating search" message/changing description text is a bit of a weird pattern that I don't know we use anywhere else

seems weird to me too - was copied directly from sitemaps, so if we change it we should do so there too.

**Comment by beaulebens** [comment]
Hah, fair enough, thanks. And we can definitely just tackle them both in a separate PR.

File: client/my-sites/site-settings/search.jsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		} = this.props;
+
+		// for now, don't even show upgrade nudge
+		if ( ! enableFeature ) {
+			return null;
+		}
+
+		return (
+			<div>
+				{ siteId && <QueryJetpackConnection siteId={ siteId } /> }
```
Comment (tyxla) [inline client/my-sites/site-settings/search.jsx]: We don't need the `<QueryJetpackConnection />` for anything here, do we?

File: client/my-sites/site-settings/search.jsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	( state ) => {
+		const site = getSelectedSite( state );
+		const siteId = getSelectedSiteId( state );
+		const isSearchEligible = site && site.plan && hasBusinessPlan( site.plan );
+
+		return {
+			siteId,
+			activatingSearchModule: !! isActivatingJetpackModule( state, siteId, 'search' ),
+			enableFeature: isSearchEligible,
+			site: getSelectedSite( state ),
```
Comment (tyxla) [inline client/my-sites/site-settings/search.jsx]: We are not using `site` anywhere, so no need to pass it as a connected prop.

File: client/my-sites/site-settings/search.jsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+			siteSlug: getSelectedSiteSlug( state ),
```
Comment (tyxla) [inline client/my-sites/site-settings/search.jsx]: We are not using `siteSlug` anywhere, so no need to pass it as a connected prop.

File: client/my-sites/site-settings/search.jsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+			siteIsJetpack: isJetpackSite( state, siteId ),
+			searchModule: getJetpackModule( state, siteId, 'search' ),
```
Comment (tyxla) [inline client/my-sites/site-settings/search.jsx]: Seems like this one is not used and can be removed as well.

**Pull request merged**